apps/nsq_to_nsq/nsq_to_nsq.go

```diff
@@ -349,37 +402,45 @@ func main() {
 		hostPool = hostpool.NewEpsilonGreedy(destNsqdTCPAddrs, 0, &hostpool.LinearEpsilonValueCalculator{})
 	}
 
-	handler := &PublishHandler{
-		addresses:        destNsqdTCPAddrs,
-		producers:        producers,
-		mode:             selectedMode,
-		hostPool:         hostPool,
-		respChan:         make(chan *nsq.ProducerTransaction, len(destNsqdTCPAddrs)),
-		perAddressStatus: perAddressStatus,
-		timermetrics:     timer_metrics.NewTimerMetrics(*statusEvery, "[aggregate]:"),
-	}
-	consumer.AddConcurrentHandlers(handler, len(destNsqdTCPAddrs))
+	consumerList := initConsumerAndHandler(cCfg, producers, selectedMode, hostPool, perAddressStatus)
 
-	for i := 0; i < len(destNsqdTCPAddrs); i++ {
-		go handler.responder()
+	for _, consumer := range consumerList {
+		err := consumer.ConnectToNSQDs(nsqdTCPAddrs)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
-	err = consumer.ConnectToNSQDs(nsqdTCPAddrs)
-	if err != nil {
-		log.Fatal(err)
+	for _, consumer := range consumerList {
+		err := consumer.ConnectToNSQLookupds(lookupdHTTPAddrs)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
 
-	err = consumer.ConnectToNSQLookupds(lookupdHTTPAddrs)
-	if err != nil {
-		log.Fatal(err)
+	aggregateStopChan := make(chan bool)
+
+	for _, consumer := range consumerList {
+		go func(consumer *nsq.Consumer, aggregateStopChan chan<- bool) {
```

Comment:
I suggest looking into [`sync.WaitGroup`](https://golang.org/pkg/sync/#WaitGroup) to handle this cleanup instead of this loop/goroutine/channel approach.